kubespawner/spawner.py

```diff
@@ -49,7 +54,9 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # By now, all the traitlets have been set, so we can use them to compute
         # other attributes
-        self.executor = SingletonExecutor.instance(max_workers=self.k8s_api_threadpool_workers)
+        # FIXME: Can't actually use the .instance here and in IngressReflector, traitlets will throw
```

Comment:
@minrk can you suggest a clean way to do this? I want the executor to be a singleton since we want all spawners to share the same threadpool.

Reply:
The singleton classes are mainly useful for registering global (process-wide) instances that aren't attached to anything, like IPython itself, or the current Application. If it's a class-global, a normal class attribute ought to work better:

```python
# class-level
executor = None

def __init__(...):
    ....
    if self.executor is None:
        # instantiate executor at the class level
        self.__class__.executor = ThreadPoolExecutor(max_workers...)
    # self.executor from now on is shared by all instances
```

Reply:
@minrk I've removed all places that was using trailets for singletons now!